### PR TITLE
plumbing clusterconfig file values with create cluster command and ch…

### PR DESCRIPTION
**Pull request by panktishah26**
…anges needed for creating cluster using tinkerbell provider

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/providers/tinkerbell/config/template-cp.yaml [hunk 1]
```diff
@@ -75,4 +83,5 @@ metadata:
   name:  {{.clusterName}}
   namespace: {{.eksaSystemNamespace}}
 spec:
+  imageLookupFormat: {{.BaseRegistry}}/{{.OSDistro}}-{{.OSVersion}}-kube-{{.KubernetesVersion}}.raw.gz
```
Comment (ptrivedi) [inline pkg/providers/tinkerbell/config/template-cp.yaml:86]: We will need to customize this further later, possibly make the format extension substitutable and use the imageLookupBaseRegistry field for this. Fine for now though, revisit later.

**Review by abhinavmpandey08 — COMMENTED** [review]
looks good, just a few minor comments

File: pkg/providers/tinkerbell/tinkerbell.go [hunk 2]
```diff
@@ -369,6 +383,15 @@ func buildTemplateMapCP(clusterSpec *cluster.Spec, controlPlaneMachineSpec v1alp
 		"kubeVipImage":                 "ghcr.io/kube-vip/kube-vip:latest",
 		"podCidrs":                     clusterSpec.Spec.ClusterNetwork.Pods.CidrBlocks,
 		"serviceCidrs":                 clusterSpec.Spec.ClusterNetwork.Services.CidrBlocks,
+		"BaseRegistry":                 "", // TODO: need to get this values for creating template IMAGE_URL
+		"OSDistro":                     "", // TODO: need to get this values for creating template IMAGE_URL
+		"OSVersion":                    "", // TODO: need to get this values for creating template IMAGE_URL
+		"KubernetesVersion":            "", // TODO: need to get this values for creating template IMAGE_URL
```
Comment (abhinavmpandey08) [inline pkg/providers/tinkerbell/tinkerbell.go:389]: camelCase for consistency
```suggestion
		"baseRegistry":                 "", // TODO: need to get this values for creating template IMAGE_URL
		"osDistro":                     "", // TODO: need to get this values for creating template IMAGE_URL
		"osVersion":                    "", // TODO: need to get this values for creating template IMAGE_URL
		"kubernetesVersion":            "", // TODO: need to get this values for creating template IMAGE_URL
```

File: pkg/providers/tinkerbell/tinkerbell.go [hunk 3]
```diff
@@ -132,14 +132,15 @@ func (p *tinkerbellProvider) DeleteResources(_ context.Context, _ *cluster.Spec)
...
+		return fmt.Errorf("failed setup and validations: %v", err)
+	}
 	logger.Info("Warning: The tinkerbell infrastructure provider is still in development and should not be used in production")
 	p.controlPlaneSshAuthKey = p.machineConfigs[p.clusterConfig.Spec.ControlPlaneConfiguration.MachineGroupRef.Name].Spec.Users[0].SshAuthorizedKeys[0]
 	p.workerSshAuthKey = p.machineConfigs[p.clusterConfig.Spec.WorkerNodeGroupConfigurations[0].MachineGroupRef.Name].Spec.Users[0].SshAuthorizedKeys[0]
 	// TODO: Add more validations
-	err := p.validateEnv(ctx)
-	if err != nil {
-		return fmt.Errorf("failed setup and validations: %v", err)
-	}
```
Comment (abhinavmpandey08) [inline pkg/providers/tinkerbell/tinkerbell.go:142]: We should remove this from SetupAndValidateDeleteCluster as well

**Pull request merged**